python/ray/data/impl/compute.py

```
@@ -105,13 +106,29 @@ class ActorPoolStrategy(ComputeStrategy):
     For a fixed-sized pool of size ``n``, specify ``compute=ActorPoolStrategy(n, n)``.
     """
 
-    def __init__(self, min_size: int = 1, max_size: Optional[int] = None):
+    def __init__(
+        self,
+        min_size: int = 1,
+        max_size: Optional[int] = None,
+        max_tasks_in_flight: Optional[int] = 2,
+    ):
+        """Construct ActorPoolStrategy for a Dataset transform.
+
+        Args:
+            min_size: The minimize size of the actor pool.
+            max_size: The maximum size of the actor pool.
+            max_tasks_in_flight: The maximum number of tasks to concurrently send to a
+                single actor worker. Increasing this will increase opportunities for
+                pipelining task dependency prefetching with computation and avoiding
+                actor startup delays, but will also increase queueing delay.
+        """
         if min_size < 1:
             raise ValueError("min_size must be > 1", min_size)
         if max_size is not None and min_size > max_size:
             raise ValueError("min_size must be <= max_size", min_size, max_size)
         self.min_size = min_size
         self.max_size = max_size or float("inf")
+        self.max_tasks_in_flight = max_tasks_in_flight
```

Could we raise an error if max_tasks_in_flight is less than 1?